### ISSUE TITLE
LoggerListener should log batch sizes for BatchSingle and BatchMultiple

### ISSUE THREAD

**Issue by AnghelLeonard** [jOOQ/jOOQ#11467]
### Use case:
Fine-tuning batching.

### Possible solution you'd like to see:

I was expecting in batching-log (e.g., when using `batch()`):

- a clear signal that baching is really used (maybe this is a bad habit from Hibernate, which can disable batching in certain scenarios and there is no signal other than logs - you think it batches, but is not)
- a clear demarcation of batches by size and type (e.g., to check that batch size work)
- the total number of executed batches (e.g., to check that the expected number is the same as the executed number and apply further optimizations to reduce the number of batches)

PS: When using `BatchedConnection` the log is much more verbose and clear. I see messages such as:

```txt
Batched 1 times:....
Batched 2 times:....
Batched 3 times:....
...
Executed with 6 batched items:...
```
These come from `o.j.tools.jdbc.BatchedPreparedStatement`

### Possible workarounds:
Activate JDBC logging, but this is JDBC-driver specific.

### Versions:

- jOOQ:3.14.7
- Java:14
- Database (include vendor):-
- OS:-
- JDBC Driver (include name if inofficial driver):-

**Comment by lukaseder** [comment]
Thanks a lot for your feature request. I agree, the logging can be improved in this area, and probably in other areas as well.

**Comment by lukaseder** [comment]
Looking into this now. There are 2 types of batches:

- `BatchSingle` (a single prepared statement with multiple bind variable sets)
- `BatchMultiple` (multiple static statements)

In both cases, `LoggerListener` already debug-logs something. In the `BatchSingle` case, only the query is logged. In the `BatchMultiple` case, each individual statement is logged.

Both types should log an overview line that counts the statements. In the `BatchMultiple` case, the statements are already available in `ExecuteContext::batchSQL`, so this is easy. In the `BatchSingle` case, we could collect `bindEnd()` events and log the batch size in the `executeStart` event.

**Comment by lukaseder** [comment]
Regarding your bullets

> * a clear signal that baching is really used (maybe this is a bad habit from Hibernate, which can disable batching in certain scenarios and there is no signal other than logs - you think it batches, but is not)

This is already happening in the `LoggerListener`. There's a message "Executing batch query", though there's a bug that logs only the first `BatchMultiple` query, not the others. For `BatchSingle`, the behaviour is correct.

(Of course, this is assuming that the `LoggerListener` is active, and `DEBUG` logging is enabled)

> * a clear demarcation of batches by size and type (e.g., to check that batch size work)

The size will be added.

> * the total number of executed batches (e.g., to check that the expected number is the same as the executed number and apply further optimizations to reduce the number of batches)

THe number of executed batches is irrelevant to an individual batch. Did you mean for this to improve also `Loader` API output? If so, would you mind creating a new, separate issue?

In summary, this issue here:

- Fixes a flaw related to `BatchMultiple` (only first statement logged)
- Adds a "Batch size" log entry
